queries.sql

```diff
@@ -7,4 +7,74 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts FROM animals WHERE weight > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight >= 10.4 AND weight <= 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight >= 10.4 AND weight <= 17.3;
+
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+ROLLBACK;
+
+BEGIN;
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+COMMIT;
+
+BEGIN;
+
+DELETE FROM animals;
+SELECT * FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+BEGIN;
+
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT sp1;
+
+UPDATE animals
+SET weight = weight * -1;
+
+ROLLBACK TO sp1;
+
+UPDATE animals
+SET weight = weight * -1
+WHERE weight < 0;
+
+COMMIT;
+
+SELECT COUNT(*) FROM animals;
+
+SELECT COUNT(*) FROM animals
+WHERE escape_attempts = 0;
+
+SELECT AVG(weight) FROM animals;
+
+SELECT AVG(escape_attempts) as Avg_Escape_Attempts,
+neutered
+FROM animals
+GROUP BY neutered;
+
+SELECT species,
+MIN(weight) as Min_Weight,
+MAX(weight) as Max_Weight
+FROM animals
+GROUP BY species;
+
+SELECT species,
+AVG(escape_attempts) as Avg_Escape_Attempts
+FROM animals
+WHERE
+date_of_birth >= '1990-01-01' AND
+date_of_birth <= '2000-01-01'
```

Comment:
- [ ] Here you should select the last day of the year to capture the whole year, kindly consider fixing this so the data is 👉🏼 `2000/12/31`

Reply:
I appreciate your suggestion, @Ismailco. But the comparison I used here is 'greater than or equal to' (>=), which represents the exact same thing as what you suggest. I like it better this way.